code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -91,12 +91,16 @@
 /obj/item/borg/upgrade/vtec/action(var/mob/living/silicon/robot/R)
 	if(..())
 		return
-	if(R.speed < 0)
+	if(R.health != R.maxHealth)
+		to_chat(R, "<span class='notice'>System damaged, unable to install upgrade!</span>")
+		to_chat(usr, "<span class='notice'>VTEC is unable to be engaged when robot is damaged, repair it first!</span>")
```

Comment:
Reminder: this still needs to be added to the changelog.

Reply:
Right, put it in the PR description, not CL, adding it now